### PR TITLE
Added section about interrupts, closes #98

### LINKED ISSUE #98: document #[interrupt]
svd2rust 0.14.0 is out so this is now unblocked

### PR DESCRIPTION
**Pull request by therealprof**
Signed-off-by: Daniel Egger <daniel@eggers-club.de>

### CODE REVIEW

File: src/start/interrupts.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+Similar to exeption handlers it is also possible to declare `static mut`
+variables inside the interrupt handlers for *safe* state keeping.
+
+``` rust,ignore
+#[interrupt]
+fn TIM2() {
+    static mut COUNT: u32 = 0;
+
+    // `COUNT` has type `&mut u32` and it's safe to use
```
Comment (korken89) [inline src/start/interrupts.md]: This is in contradiction with the definition, as the interrupt macro does this conversion for us.
I would expand this comment or discuss to not have readers be confused.

File: src/start/interrupts.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+Interrupt handlers look like plain functions (except for the lack of arguments)
+similar to exception handlers. However they can not be called directly by other
+parts of the firmware due to the special calling conventions. It is however
+possible to generate interrupt requests in software to trigger a diversion to
+to the interrupt handler.
+
+Similar to exeption handlers it is also possible to declare `static mut`
+variables inside the interrupt handlers for *safe* state keeping.
```
[the remark this answers is not in the document]
Comment (therealprof) [reply src/start/interrupts.md]: @korken89 That explanation is in the exception section already and I didn't want to copy half of the section in verbatim. Maybe add an anchor and a link?

File: src/start/interrupts.md (hunk 1 above)
Comment (therealprof) [reply src/start/interrupts.md]: Again, that is taken from the exception section. ;)

Comment (korken89) [reply src/start/interrupts.md]: I think a reference will do perfectly! Just to minimize the chance of misunderstanding :)

**Comment by therealprof** [comment]
@korken89 Added the link.

**Pull request merged**